src/home/components/PetWeightPage/petWeightPage.spec.js

```diff
@@ -1,16 +1,188 @@
+import RangeSlider from '../../../components/RangeSlider';
+import TextInput from '../../../components/TextInput';
+import RadioButton from '../../../components/RadioButton';
+import Button from '../../../components/Button';
+
+let instance;
+
+vi.mock('../../../components/UploadImage', () => ({
+  __esModule: true,
+  default: vi.fn().mockImplementation(() => ({
+    mount: vi.fn(),
+    selected: {
+      get: vi.fn().mockReturnValue({
+        classList: {
+          add: vi.fn(),
+        },
+      }),
+    },
+  })),
+}));
+
+vi.mock('../../../components/RangeSlider', () => ({
+  __esModule: true,
+  default: vi.fn().mockImplementation(() => ({
+    mount: vi.fn(),
+    setValue: vi.fn(),
+    listen: vi.fn(),
+    selected: {
+      get: vi.fn().mockReturnValue({
+        classList: {
+          add: vi.fn(),
+        },
+      }),
+    },
+  })),
+}));
+
+vi.mock('../../../components/TextInput', () => ({
+  __esModule: true,
+  default: vi.fn().mockImplementation(() => ({
+    mount: vi.fn(),
+    setValue: vi.fn(),
+    listen: vi.fn(),
+    selected: {
+      get: vi.fn().mockReturnValue({
+        classList: {
+          add: vi.fn(),
+        },
+      }),
+    },
+  })),
+}));
+
+vi.mock('../../../components/RadioButton', () => ({
+  __esModule: true,
+  default: vi.fn().mockImplementation(() => ({
+    mount: vi.fn(),
+    isChecked: vi.fn().mockImplementation(function isChecked() {
+      return this === instance.radioKG;
+    }),
+    getValue: vi.fn().mockImplementation(function getValue() {
+      return this === instance.radioKG ? 'kg' : 'lb';
+    }),
+    selected: {
+      get: vi.fn().mockReturnValue({
+        classList: {
+          add: vi.fn(),
+        },
+      }),
+    },
+  })),
+}));
+
+vi.mock('../../../components/Button', () => ({
+  __esModule: true,
+  default: vi.fn().mockImplementation(() => ({
+    mount: vi.fn(),
+    listen: vi.fn(),
+    selected: {
+      get: vi.fn().mockReturnValue({
+        classList: {
+          add: vi.fn(),
+        },
+      }),
+    },
+  })),
+}));
 
 const propsMock = {
   petPhoto: 'https://via.placeholder.com/150',
 };
 
 describe('PetWeightPage', () => {
+  beforeEach(() => {
+    instance = new PetWeightPage(propsMock);
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
   it('is a Function', () => {
     expect(PetWeightPage).toBeInstanceOf(Function);
   });
 
-  it.skip('returns an object', () => {
-    expect(new PetWeightPage(propsMock)).toBeInstanceOf(Object);
+  it('returns an object', () => {
+    expect(instance).toBeInstanceOf(Object);
+  });
+
+  it('initializes components correctly', () => {
+    expect(UploadImage).toHaveBeenCalledTimes(1);
+    expect(RangeSlider).toHaveBeenCalledTimes(1);
+    expect(TextInput).toHaveBeenCalledTimes(1);
+    expect(RadioButton).toHaveBeenCalledTimes(2);
+    expect(Button).toHaveBeenCalledTimes(1);
+  });
+
+  it('sets the initial pet photo', () => {
+    expect(UploadImage).toHaveBeenCalledWith();
```

Comment:
Evitar testes com `toHaveBeenCalled` e `toHaveBeenCalledWith`, queremos sempre garantir comportamento, não coisas internas do código

No caso, queremos que, apos adicionar uma foto e colocar o peso, queremos que no evento de submit recebemos a nova foto e o peso selecionado e é isso que queremos testar. To be called with é um teste de implementação, oque nos queremos é teste de comportamento, do modo como o componente é usado em si!